R/MEND2.R

```diff
@@ -0,0 +1,225 @@
+    dM <- (1 - f.d) * (fluxes$F2p1() + fluxes$F2p2()) - fluxes$F3()
+    # Q = active layer of MOC (2 pools)
+    dQ1 <- fluxes$F6d1() - fluxes$F7d1()
+    dQ2 <- fluxes$F6d2() - fluxes$F7d2()
+    # B = microbial biomass carbon
+    dB <- E.c1*fluxes$F1d1() + E.c2*fluxes$F1d2() -  fluxes$F8() - (fluxes$F9.em() + fluxes$F9.ep1() +  fluxes$F9.ep2())
+    # D = dissolved organic carbon (2 pools)
+    dD1 <- I.d1 +  g.d * fluxes$F8() +  fluxes$F3()  + 0.5 * f.d * (fluxes$F2p1() + fluxes$F2p2()) + (fluxes$F10.em() + fluxes$F10.ep1() + + fluxes$F10.ep2()) - fluxes$F1d1() - (fluxes$F6d1() - fluxes$F7d1())
+    dD2 <- I.d2 +  0.5 * f.d*  (fluxes$F2p1() + fluxes$F2p2()) - fluxes$F1d2() - (fluxes$F6d2() - fluxes$F7d2())
+    # EP = carbon stored as extracellular enzymes (2 pools)
+    dEP1 <- fluxes$F9.ep1() - fluxes$F10.ep1()
+    dEP2 <- fluxes$F9.ep2() - fluxes$F10.ep2()
+    # EM = carbon stored as extracellular enzymes
+    dEM <- fluxes$F9.em() - fluxes$F10.em()
+    # IC = inorganic carbon (CO2)
+    dIC <-  (1 - E.c1) *  fluxes$F1d1() + (1 - E.c2) *  fluxes$F1d2()
+    # Tot = the total carbon pool
+    dTot <- I.p1 + I.p2 + I.d1 + I.d2 - (1 - E.c1)*fluxes$F1d1() - (1 - E.c2)*fluxes$F1d2()
+
+    # Return outputs
+    list(c(dP1, dP2, dM, dB, dD1, dD2, dQ1, dQ2, dEP1, dEP2, dEM, dIC, dTot))
+  })
+}
+
+
+
+
+#' Define the 2 P pool MEND fluxes
+#'
+#' \code{MEND2_fluxes} Defines a system of equations that
+#' describe the state of the fluxes between the pools from the 2 pool version of
+#' \href{https://doi.org/10.1890/12-0681.1}{Wang et al. 2013}. By retruning a list of named functions.
+#'
+#' @param state A numeric vector of the different MEND carbon pool states.
+#' @param parms A data frame of the parameters.
+#' @return A list of functions that calculate the fluxes between the MEND 2 carbon pools.
+#' @family MEND2
+#' @family carbon flux functions
+#' @importFrom assertthat assert_that has_name
+#' @noRd
+MEND2_fluxes <- function(state, parms){
+
+  # Check inputs
+  assert_that(has_name(x = state, which = c("P1", "P2", "M", "B", "D1", "D2", "Q1",
+                                            "Q2", "EP1", "EP2", "EM", "IC", "Tot")))
+  assert_that(has_name(x = parms, which = c("parameter", "description", "units", "value")))
+  assert_that(has_name(x = parms, which = c("parameter",  "value")))
+
+  req <- c('E.c1', 'E.c2', 'V.d1', 'V.d2', 'K.d1', 'K.d2', 'V.p1', 'V.p2', 'K.p1', 'K.p2', 'K.m', 'K.ads',
+           'Q.max1', 'Q.max2', 'K.des', 'p.ep', 'p.em', 'r.ep',  'r.em')
+  missing <- setdiff(req, parms$parameter)
+  assert_that(length(missing) == 0, msg = paste('parms missing values for: ', paste0(missing, collapse = ', ')))
+
+
+  # Format the parameters into a vector.
+  p        <- parms$value
+  names(p) <- parms$parameter
+
+  with(as.list(c(state, p)), {
+
+    fxn_list <- list(
+      "F1d1" = function(){
+        # Microbial uptake from D1 pool
+        V.d1 * B * D1 /( K.d1 + D1)
+      },
```

Comment:
What happened to the 1/Ec & the mr term? 

![](https://esajournals.onlinelibrary.wiley.com/cms/asset/acddee9d-24a9-4aa8-be75-86388c52e091/eap2013231255-math-0009.gif)

Reply:
I simplified the expression with maximum uptake rate (Vd1, Vd2 etc) and carbon use efficiency (Ec1 & Ec2)